File: test/detect_driver_manager.pl
#!/usr/bin/env perl

use 5.16.0;
use utf8;
use open ':encoding(utf8)', ':std';
use strict;
use warnings;
use DBI;

eval {
    my $dbh = DBI->connect('dbi:ODBC:DSN=__nonexistent_dsn__', '', '', { RaiseError => 1, PrintError => 0 });
};

if ($@ =~ /\[([^\[\]]+)\]\[Driver Manager\]/) {
    print "$1\n";
}
elsif ($@) {
    die $@;
}

File: test/detect_driver_manager.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import re
import pyodbc

try:
    connection = pyodbc.connect("DSN=__nonexistent_dsn__")
except pyodbc.Error as error:
    result = re.search(r"\[([^\[\]]+)]\[Driver Manager]", str(error))
    if result and len(result.groups()) >= 1:
        print(result.group(1))
    else:
        raise
